src/hdmf/common/io/table.py

```diff
@@ -43,3 +44,63 @@ def get_attr_value(self, **kwargs):
             elif spec.data_type_inc == 'VectorIndex':
                 attr_value = container[spec.name]
         return attr_value
+
+
+class DynamicTableGenerator(CustomClassGenerator):
+
+    @classmethod
+    def update_cls_args(cls, classdict, bases, not_inherited_fields, spec, type_map):
+        """Update the given class dict and base classes if there is a field spec with quantity * or +
+        :param classdict: The dict to update with __clsconf__ if applicable
+        :param bases: The list of base classes to update if applicable
+        :param not_inherited_fields: Dict of additional fields that are not in the base class
```

Comment:
by `base class` do you mean the base data_type/included spec?

Reply:
I mean the parent type. For DynamicTable, that is Container. So `not_inherited_fields` contains specs for attributes 'description' and 'colnames', and datasets 'id', and included 'VectorData'.